lib/ASRouterTriggerListeners.jsm

```diff
@@ -10,35 +10,15 @@ ChromeUtils.defineModuleGetter(
   "PrivateBrowsingUtils",
   "resource://gre/modules/PrivateBrowsingUtils.jsm"
 );
+ChromeUtils.defineModuleGetter(
+  this,
+  "EveryWindow",
+  "resource:///modules/EveryWindow.jsm"
+);
 
 const FEW_MINUTES = 15 * 60 * 1000; // 15 mins
 const MATCH_PATTERN_OPTIONS = { ignorePath: true };
 
-/**
- * Wait for browser startup to finish to avoid accessing uninitialized
- * properties
- */
-async function checkStartupFinished(win) {
```

Comment:
Does `EveryWindow.registerCallback`remove the need to have to do this essentially?

Reply:
Yes. The callback is [called only after delayedStartup](https://searchfox.org/mozilla-central/rev/588814f2edddf0e132d77d326ddae50911e8bad1/browser/modules/EveryWindow.jsm#39-41) finished.